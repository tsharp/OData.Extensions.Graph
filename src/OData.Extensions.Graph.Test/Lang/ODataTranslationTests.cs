using Microsoft.OData;
using OData.Extensions.Graph.Lang;
using Snapshooter.Xunit;
using Xunit;

namespace OData.Extensions.Graph.Test.Lang
{
    public class ODataTranslationTests
    {
        [Fact]
        public static void EmptySelectClause()
        {
            // Arrange
            var translator = new QueryTranslator(Common.GetEdmModel());

            // Act

            // Assert
            Assert.Throws<ODataException>(() => translator.Translate("/user?$select"));
            Assert.Throws<ODataException>(() => translator.Translate("/user?$expand"));
            Assert.Throws<ODataException>(() => translator.Translate("/user?$select&$expand"));
        }

        [Fact]
        public static void EntitySetTranslation()
        {
            // Arrange
            var translator = new QueryTranslator(Common.GetEdmModel());

            // Act
            var basic_selectUserId = translator.Translate("/user?$select=Id");

            // Assert
            Snapshot.Match(basic_selectUserId);
        }
    }
}
