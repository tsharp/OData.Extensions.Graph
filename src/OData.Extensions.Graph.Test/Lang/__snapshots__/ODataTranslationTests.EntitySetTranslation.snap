{
  "DocumentNode": {
    "Kind": "Document",
    "Location": null,
    "Definitions": [
      {
        "Kind": "OperationDefinition",
        "Location": null,
        "Name": null,
        "Operation": "Query",
        "VariableDefinitions": [],
        "Directives": [],
        "SelectionSet": {
          "Kind": "SelectionSet",
          "Location": null,
          "Selections": [
            {
              "Kind": "Field",
              "Alias": null,
              "Arguments": [],
              "Required": null,
              "SelectionSet": {
                "Kind": "SelectionSet",
                "Location": null,
                "Selections": [
                  {
                    "Kind": "Field",
                    "Alias": null,
                    "Arguments": [],
                    "Required": null,
                    "SelectionSet": null,
                    "Location": null,
                    "Name": {
                      "Kind": "Name",
                      "Location": null,
                      "Value": "Id"
                    },
                    "Directives": []
                  }
                ]
              },
              "Location": null,
              "Name": {
                "Kind": "Name",
                "Location": null,
                "Value": "user"
              },
              "Directives": []
            }
          ]
        }
      }
    ]
  },
  "PathSegment": "user"
}
